runtime/jvm/src/main/kotlin/kotlinx/serialization/Resolving.kt

```diff
@@ -47,38 +48,76 @@ private fun mapJavaClassNameToKotlin(s: String): String = when(s) {
 }
 
 fun <E> serializerByValue(value: E, context: SerialContext? = null): KSerializer<E> {
-    val klass = (value as? Any)?.javaClass?.kotlin ?: throw SerializationException("Cannot determine class for value $value")
+    val klass =
+        (value as? Any)?.javaClass?.kotlin ?: throw SerializationException("Cannot determine class for value $value")
     return serializerByClass(klass, context)
 }
 
-fun <E> serializerBySerialDescClassname(className: String, context: SerialContext? = null): KSerializer<E> = SerialCache.lookupSerializer(className, context = context)
+fun <E> serializerBySerialDescClassname(className: String, context: SerialContext? = null): KSerializer<E> =
+    SerialCache.lookupSerializer(className, context = context)
+
+fun <E> serializerByClass(klass: KClass<*>, context: SerialContext? = null): KSerializer<E> =
+    SerialCache.lookupSerializer(mapJavaClassNameToKotlin(klass.java.canonicalName ?: ""), klass, context)
 
-fun <E> serializerByClass(klass: KClass<*>, context: SerialContext? = null): KSerializer<E> = SerialCache.lookupSerializer(mapJavaClassNameToKotlin(klass.java.canonicalName ?: ""), klass, context)
+open class TypeBase<T>
```

Comment:
I think this declaration could be `internal` instead of `public`, because there is no need for this class outside of this file. However, because it is used in the inline function, you have to mark it as `@PublishedApi` so the compiler will not complain about it. See how it's done in ktor: https://github.com/ktorio/ktor/blob/6c6965c9cafdcc1f6fae2d5635b0ffcf2839afa3/ktor-client/ktor-client-core/ktor-client-core-jvm/src/io/ktor/client/call/TypeInfoJvm.kt#L9